client/src/components/ExternalLinks.tsx

```diff
@@ -0,0 +1,262 @@
+    displayTitle = iconAfter ? (
+      <span className="btn-icon-text-after">
+        {title}
+        {icon}
+      </span>
+    ) : (
+      <span className="btn-icon-text">
+        {icon}
+        {title}
+      </span>
+    );
+  }
+  const myProps = { ...props, title: displayTitle };
+  if (role === "link" || role === "text")
+    return <ExternalLinkText {...myProps} />;
+  return <ExternalLinkButton {...myProps} />;
+}
+
+interface ExternalDocsLinkProps {
+  children?: React.ReactNode | null;
+  /** [Optional] Any classes to add, e.g., 'nav-link' or 'dropdown-item' */
+  className?: string;
+  /** Show the icon to indicate an external link if true (default false) */
+  showLinkIcon?: boolean;
+  /** The text to show for the link */
+  title: string;
+  /** The URL to link to */
+  url: string;
+}
+/**
+ * Link to external URL, with the role as text.
+ */
+export function ExternalDocsLink(props: ExternalDocsLinkProps) {
+  const role = "link";
+  return <ExternalLink role={role} {...props} />;
+}
+
+interface IconLinkProps {
+  /** An icon to display */
+  icon: React.ReactNode;
+  /** The URL to link to */
+  to: string;
+  /** The text of the tooltip */
+  tooltip: string;
+}
+/**
+ * IconLink
+ * Internal application link that is shown as a font-awesome icon
+ */
+export function IconLink(props: IconLinkProps) {
+  const ref = useRef(null);
+
+  return (
+    <span>
+      <Link ref={ref} to={props.to}>
+        {props.icon}
+      </Link>
+      <UncontrolledTooltip target={ref}>{props.tooltip}</UncontrolledTooltip>
+    </span>
+  );
+}
+
+function ExternalIconLinkWithTooltip(props: ExternalIconLinkProps) {
+  const ref = useRef(null);
+
+  return (
+    <span>
+      <a
+        href={props.url}
+        role="button"
+        className={cx("btn-icon-text", "icon-link", props.className)}
+        ref={ref}
+        target="_blank"
+        rel="noreferrer noopener"
+      >
+        {props.icon}
+        {props.title}
+      </a>
+      <UncontrolledTooltip target={ref}>{props.tooltip}</UncontrolledTooltip>
+    </span>
+  );
+}
+
+function ExternalIconLinkWithoutTooltip({
+  className,
+  icon,
+  title,
+  url,
+}: ExternalIconLinkProps) {
+  return (
+    <a
+      href={url}
+      target="_blank"
+      rel="noreferrer noopener"
+      className={cx("btn-icon-text", className)}
+    >
+      {icon}
+      {title}
+    </a>
+  );
+}
+
+interface ExternalIconLinkProps {
+  /** [Optional] Any classes to add, e.g., 'nav-link' or 'dropdown-item' */
+  className?: string;
+  /** An icon to display */
+  icon: React.ReactNode;
+  /** The text to show for the link */
+  title: string;
+  /** The text of the tooltip */
+  tooltip?: string;
+  /** The URL to link to */
+  url: string;
+}
+export function ExternalIconLink(props: ExternalIconLinkProps) {
+  return props.tooltip ? (
+    <ExternalIconLinkWithTooltip {...props} />
+  ) : (
+    <ExternalIconLinkWithoutTooltip {...props} />
+  );
+}
```

Comment:
We can handle tooltips just fine in one function here. Also, increase flexibility by using `ReactNode`.

```suggestion
interface ExternalIconLinkProps {
  /** [Optional] Any classes to add, e.g., 'nav-link' or 'dropdown-item' */
  className?: string;
  /** An icon to display */
  icon: ReactNode;
  /** The text to show for the link */
  title: ReactNode;
  /** The text of the tooltip */
  tooltip?: ReactNode;
  /** The URL to link to */
  url: string;
}
export function ExternalIconLink({
  className,
  icon,
  title,
  tooltip,
  url,
}: ExternalIconLinkProps) {
  const ref = useRef<HTMLAnchorElement>(null);

  return (
    <>
      <a
        href={url}
        role="button"
        className={cx("btn-icon-text", "icon-link", className)}
        ref={ref}
        target="_blank"
        rel="noreferrer noopener"
      >
        {icon}
        {title}
      </a>
      {tooltip && (
        <UncontrolledTooltip target={ref}>{tooltip}</UncontrolledTooltip>
      )}
    </>
  );
}
```